crates/kitsune_p2p/kitsune_p2p/src/test_util/harness_agent.rs

```diff
@@ -79,6 +85,21 @@ impl ghost_actor::GhostControlHandler for AgentHarness {}
 impl ghost_actor::GhostHandler<HarnessAgentControl> for AgentHarness {}
 
 impl HarnessAgentControlHandler for AgentHarness {
+    fn handle_dump_agent_info(
+        &mut self,
+    ) -> HarnessAgentControlHandlerResult<Vec<Arc<AgentInfoSigned>>> {
+        let all = self.agent_store.values().map(|a| a.clone()).collect();
+        Ok(async move { Ok(all) }.boxed().into())
+    }
```

Comment:
I think we both wrote basically the same function :laughing: 
See `handle_dump_local_peer_data()`